src/main/java/ome/codecs/JPEGCodec.java

```diff
@@ -76,13 +80,19 @@ public byte[] compress(byte[] data, CodecOptions options)
       throw new CodecException("> 8 bit data cannot be compressed with JPEG.");
     }
 
-    ByteArrayOutputStream out = new ByteArrayOutputStream();
+    ByteArrayOutputStream out = new ByteArrayOutputStream(Math.max(1024, data.length / 4));
```

Comment:
Might be worth limiting this a bit further - in the worst case where `data.length` is `Integer.MAX_VALUE`, this would set the output stream's length to `536870911`.